util/reggen/validate.py

```diff
@@ -662,21 +664,67 @@ def resolve_value(entry, param_list):
     return int(val), err
 
 
-# only supports subtractions right now
-def resolve_bitfield(entry, param_list):
-    lead_term = True
+# resolve expression
+def resolve_expression(entry, ops, param_list):
+    log.info("Evaluating entry {}".format(entry))
+
     val = 0
     error = 0
-    terms = entry.replace(" ", "").split('-')
 
-    for i in terms:
-        if lead_term:
-            lead_term = False
-            val, err = resolve_value(i, param_list)
+    # construct the pattern to search
+    operators = ''
```

Comment:
it feels like there should be a much smarter way to construct the pattern...does anyone know..?